src/components/Modal.ts

```diff
@@ -0,0 +1,32 @@
+import { FC } from '..';
+import { ContainerProps } from './ContainerProps';
+import { Block } from './Block';
+import { KnownBlock } from '@slack/types';
+import { ModalType } from './Text';
+
+export interface ModaleProps
+  extends ContainerProps<ReturnType<Block<any, KnownBlock>>> {
+  callbackId: string;
+  title: string;
+  type: ModalType;
+}
+
+export interface ModalSpec {
```

Comment:
this already exists as `import { View } from '@slack/types'`

```
export interface View {
    title: PlainTextElement;
    type: 'modal';
    blocks: (KnownBlock | Block)[];
    callback_id?: string;
    close?: PlainTextElement;
    submit?: PlainTextElement;
    private_metadata?: string;
    clear_on_close?: boolean;
    notify_on_close?: boolean;
}
```
- https://github.com/slackapi/node-slack-sdk/blob/master/packages/types/src/index.ts#L35

Reply:
you could always use `Pick<View, 'foo' | 'bar' | etc...>` if you don't want the whole type

Reply:
good idea. i tried that but the fact that i'm exporting/using this externally might be a 

**suggested change?:**
```
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ModalSpec
  extends Pick<View, 'type' | 'callback_id' | 'blocks' | 'title'> {}
```

**error:**
```
Type '({ children, callbackId, title, type, }: Props<ModalProps>) => { type: ModalType; callback_id: string; blocks: any[]; title: { type: "plain_text"; text: string; }; }' is not assignable to type 'FC<ModalProps, ModalSpec>'.
  Call signature return types '{ type: ModalType; callback_id: string; blocks: any[]; title: { type: "plain_text"; text: string; }; }' and 'ModalSpec' are incompatible.
    The types of 'type' are incompatible between these types.
      Type 'ModalType' is not assignable to type '"modal"'.
        Type '"home"' is not assignable to type '"modal"'.ts(2322)
```

Reply:
yeah i think if you removed your `ModalType` enum and just used literal string `"modal"` you'd have better luck...  in any event, `home` isn't really a MODAL. it's a type of `View`, for sure, but you're creating specific overrides of `View` because you want to make `Modal` a first class type. 

```
export interface ModalSpec
  extends Pick<View, 'callback_id' | 'blocks' | 'title'> {
  type: 'modal'
}
```

this might get you what you need?

Reply:
also you won't have to ts-ignore that.

PSFYI

if you really ahd an empty interface, you could just do a type alias:

```
type ModalSpec = Pick<View, 'type' | 'callback_id' | 'blocks' | 'title'>;
```

instead of extending the interface.